src/main/java/org/example/AutoSave.java

```diff
@@ -0,0 +1,24 @@
+package org.example;
+
+import org.example.services.StateStorageService;
+
+public class AutoSave implements Runnable {
+    private final StateStorageService storage;
+
+    public AutoSave(StateStorageService storage) {
+        this.storage = storage;
+    }
+
+    @Override
+    public void run() {
+        while (!Thread.currentThread().isInterrupted()) {
+            storage.save();
+            try {
+                Thread.sleep(3000);
+            } catch (InterruptedException ex) {
+                Thread.currentThread().interrupt();
+                break;
+            }
+        }
+    }
```

Comment:
it is not a good idea, because if an app is on stand-by, (nothing will depend on it or we will not add or remove pets) we gonna spend some resources on writing the same date to the file. In smaller apps, it is not a problem but in a big scale, it will lead to loss in performance